### ISSUE TITLE
mid3v2 bugs?

### ISSUE THREAD

**Issue by drone1** [quodlibet/mutagen#346]
I had to make a few changes to this script in order to get it to write frames to my test AIFF and MP3 files. I'm sorry I do not have time to thoroughly test or submit a PR but wanted to let you know. 

In `write_files()`, I had to do:

   `try: id3 = mutagen.File(filename).tags`

...rather than of: 

   `try: id3 = mutagen.id3.ID3(filename)`

Then, later in write_files() calls id3.save(filename), which works fine. I mention this because I made the same change above (from `write_files()`) to `delete_frames()`, but also had to change the `id3.save()` line to `id3.save(filename)` – and now everything works well for me. 

Thanks for building this out! Very useful.

**Comment by drone1** [comment]
Looks like I had naively been looking at the wrong code after a Google search – I was looking at someone's fork of mutagen (https://github.com/mdhiggins/mutagen).

However, I will say that if I run a simple command with my mid3v2 like so:

   `~/Library/Python/2.7/bin/mid3v2 -a "TEST" test.aif`

It seems to destroy the test.aif file, making it unplayable.

**Comment by drone1** [comment]
`mutagen-inspect test.aif` yields:

```
-- test.aif
can't sync to MPEG frame

```

**Comment by lazka** [comment]
mid3v2 can only handle "normal" id3 files, not file formats like aiff which embed id3 tags in their own file structure.

**Comment by drone1** [comment]
I see. Any reason not support AIFF as well?

The code above allows the underlying file to instantiate its own `tags`
(for AIFF, a `_IFFID3` instance, which works with its structure of course),
as opposed to forcing an `ID3` to be instantiated.

Is there a good reason not to do this?

**Comment by lazka** [comment]
> Is there a good reason not to do this?

I don't think so. Some options don't make sense like id3v1, but that's it I think.

**Comment by drone1** [comment]
Hm maybe I will do thorough testing and put up a PR.

I’m not a big python master here so before I dig I figure I will ask: how
can I clone the repo and run THAT mutagen’s mid3v2 rather than the one
installed (to which I’ve been making my changes since I didn’t see a quick
answer to this when I looked). The result I saw when I looked the other day
seemed kinda hacky and there must be a non-hacky way! Just tell me what to
google and I can find it?

**Comment by lazka** [comment]
> how can I clone the repo and run THAT mutagen’s mid3v2 rather than the one installed

``PYTHONPATH=. ./tools/mid3v2``

**Comment by drone1** [comment]
Hm, you sure? 

```
> PYTHONHOME=. ./tools/mid3v2
ImportError: No module named site
```

**Comment by lazka** [comment]
PYTHONPATH != PYTHONHOME

**Comment by drone1** [comment]
omg, my bad. Tired. That worked!
